### ISSUE TITLE
Generate native image reflect-config.json for com.google.firestore.bundle

### ISSUE THREAD

**Issue by mpeddada1** [googleapis/sdk-platform-java#2011]
We generate [configurations for `com.google.firestore.v1` classes ](https://github.com/googleapis/java-firestore/blob/main/google-cloud-firestore/src/main/resources/META-INF/native-image/com.google.cloud.firestore.v1/reflect-config.json)but not `com.google.firestore.bundle` classes. These needed to be manually added in https://github.com/googleapis/java-firestore/pull/1419.

**Comment by blakeli0** [comment]
Do we know how the `com.google.firestore.bundle` classes are added to java-firestore repo? I don't see them being passed into the generator in the [Bazel](https://github.com/googleapis/googleapis/blob/master/google/firestore/v1/BUILD.bazel#L57-L116) build file.

**Comment by mpeddada1** [comment]
The bazel rule for assembling this code is located [under the` google/firestore/bundle` directory](https://github.com/googleapis/googleapis/blob/5fcf6d86afd5b051cb19a19bce853a076ce35651/google/firestore/bundle/BUILD.bazel#L32-L51). Originally, it looks like this package was manually generated (https://github.com/googleapis/java-firestore/pull/271) in the repo and then the bazel rules were migrated over to third_party?

**Comment by blakeli0** [comment]
I see, this is a proto only module that does not involve the generator, it is generated using protoc plugin directly. If that's the case, I don't think we have a way to generate it through generator. 
Is this the only occurrence of this issue? Do you see it happening in other handwritten libraries?

**Comment by mpeddada1** [comment]
Hm I could be wrong but these classes seem to get generator updates through Owlbot (example, https://github.com/googleapis/java-firestore/pull/1304) which make me think that they aren't fully disconnected? 
 
> Is this the only occurrence of this issue? Do you see it happening in other handwritten libraries?

From what we've observed so far, Firestore seems to be the only one that has this issue. From the native image perspective, Spanner needed manually added configs but for different reason as there were explicitly using + reflectively calling some `com.google.rpc` classes  (https://github.com/googleapis/java-spanner/pull/2617)

**Comment by blakeli0** [comment]
> Hm I could be wrong but these classes seem to get generator updates through Owlbot (example, [googleapis/java-firestore#1304](https://github.com/googleapis/java-firestore/pull/1304)) which make me think that they aren't fully disconnected?

Yes they are generated into googleapis-gen through the [Bazel configurations ](https://github.com/googleapis/googleapis/blob/5fcf6d86afd5b051cb19a19bce853a076ce35651/google/firestore/bundle/BUILD.bazel#L32-L51), hence we can configure owlbot to copy the code from googleapis-gen. But the Bazel configurations only include `firestore_bundle_java_proto` and `firestore_bundle_proto`, which do not go through the gapic-generator-java code, compare to a [firestore v1 Bazel configuration](https://github.com/googleapis/googleapis/blob/5fcf6d86afd5b051cb19a19bce853a076ce35651/google/firestore/v1/BUILD.bazel#L106-L116),  that includes a `firestore_java_gapic` dependency. 

This issue suffers from the same [root cause](https://github.com/googleapis/sdk-platform-java/issues/2048#issuecomment-1739916096) as I mentioned in https://github.com/googleapis/sdk-platform-java/issues/2048, if we do prioritize these issues, I would suggest to fix https://github.com/googleapis/sdk-platform-java/issues/2048  first.
